Любительская навигация #1

IvanStrakhov opened the pull request.

IvanStrakhov committed:
Create requirements.txt

IvanStrakhov committed:
Add files via upload

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
Update README.md

IvanStrakhov committed:
reduced amount of code

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+    makePic.plt.close()
+    return data.getvalue()
+    
+app.run('0.0.0.0', 8888)
```
Обычно приложения на фласке в продакшене не запускают напрямую через run, а используют какой-нибудьт WSGI-сервер (например, см. [туториал](https://flask.palletsprojects.com/en/1.1.x/tutorial/deploy/)). Пожалуйста, заверните эту строку в `if __name__ == '__main__'`

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+        else:
+            return render_template(str_html,RA=RA,DEC=DEC,Error=Error,Name=Name,image=image,ZOOM=ZOOM,THRES=THRES)
+
+def get_ra_dec_fromString(ra_string,dec_string):
```
Эту всю функцию стоит заменить на что-нибудь готовое из astropy. Раз уж вы все равно идете в сеть, то можно воспользоваться https://docs.astropy.org/en/stable/api/astropy.coordinates.get_icrs_coordinates.html#astropy.coordinates.get_icrs_coordinates

hombit commented on Strakhov/README.md:
```diff
@@ -0,0 +1,29 @@
+Внимание!
+Программе для работы требуются следующие библиотеки: numpy, astropy,
+matplotlib, flask, cartopy.
+Cartopy, в частности, требует наличия cython, libproj-dev, proj-data, proj-bin,
```
Похоже для работы программы еще нужен scipy, добавьте в requirements.txt, пожалуйста

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+                                    return render_page(str_html='hello.html',Name=i["mainId"],image="data:image/png;base64,"+ base64.b64encode(create_figure(i['ra'],i['dec'],scale,thres)).decode("UTF-8"),RA=i['ra'],DEC=i['dec'],Error="Star with name: "+request.form["Name"]+" not found. Execution of coordinates search.\n"+"Closest star (distance={:.2f} asec): Magnitude: {:.1f}".format(float(i["dis ...
+                                    
+                            
+                        except:
```
Укажите тип исключения

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+                            return render_page(str_html='hello.html',Name=i['mainId'],image="data:image/png;base64,"+ base64.b64encode(create_figure(i['ra'],i['dec'],scale,thres)).decode("UTF-8"),RA=i['ra'],DEC=i['dec'],Error="Closest star (distance={:.2f} asec): Magnitude: {:.1f}".format(float(i["distance"])*60.0,float(i["M.V"])),ZOOM=scale,THRES=thres)
+                            
+                    
+                except:
```
Тип исключения

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+        if validate_request():
+            if request.form["Name"]!='':
+                try:
+                    resp=requests.get('http://simbad.u-strasbg.fr/simbad/sim-nameresolver?Ident='+request.form['Name']+'&data=M(V),J&output=json').json()
```
Просто используйте astroquery

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,155 @@
+app = Flask(__name__)
+
+    
+@app.route('/', methods=['POST', 'GET'])
```
Эта функция слишком большая и делает слишком много. Во-первых, стоит сделать отдельные функции для GET и POST. Во-вторых, сделайте внешнюю функцию для того, чтобы сходить в Визирь.

IvanStrakhov committed:
changes in skyStar

указаны исключения, разделены get и post, app.run - под __main__, обращение к симбаду вынесено в отдельные функции

IvanStrakhov committed:
Update requirements.txt

IvanStrakhov committed:
Update README.md

IvanStrakhov commented:
Всё исправил, кроме этого:
> Эту всю функцию стоит заменить на что-нибудь готовое из astropy. 

get_icrs_coordinates выводит координаты по имени объекта. Моя функция делает совсем другое. Я её немного подправил. Теперь она определяет единицы, в которых пользователь вводит координаты, чтобы пользователю можно было не париться и вводить их в любом из следующих форматов:

- 20 54 05.689 +37 01 17.38
- 10:12:45.3 -45:17:50
- 15h17m -11d10m
- 15h17 +89d15
- 275d11m15.6954s +17d59m59.876s
- 12.34567h -17.87654d
- 350.123456d -17.33333d
- 350.123456 -17.33333

IvanStrakhov committed:
small fix

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,172 @@
+        else:
+            return render_template(str_html,RA=RA,DEC=DEC,Error=Error,Name=Name,image=image,ZOOM=ZOOM,THRES=THRES)
+
+def get_ra_dec_units_fromString(ra_string,dec_string):
```
Этот код все еще сложно понять, наверное тут гораздо лучше подошли бы регулярные выражения или, хотя бы, замена len(str.split()) на str.count()

hombit commented on Strakhov/skyStar.py:
```diff
@@ -0,0 +1,172 @@
+    else:
+        thres=10
+        
+    if validate_request():
```
Слишком большая вложенность if-else-try, сложно понять что происходит. К тому же блоки try слишком большие, непонятно кто именно может кинуть исключение и почему

hombit commented:
Почти все эти комбинации покрываются связкой конструкторов SkyCoord:
 ```python
def get_coord_from_string(s):
    ...:     try:
    ...:         return SkyCoord(s)
    ...:     except ValueError:
    ...:         return SkyCoord.from_name(s, parse=True)

strings = '''20 54 05.689 +37 01 17.38
    ...: 10:12:45.3 -45:17:50
    ...: 15h17m -11d10m
    ...: 15h17 +89d15
    ...: 275d11m15.6954s +17d59m59.876s
    ...: 12.34567h -17.87654d
    ...: 350.123456d -17.33333d
    ...: 350.123456 -17.33333'''.splitlines()

for s in strings:
    ...:     print(s, get_coord_from_string(s))
    ...:
```

```
20 54 05.689 +37 01 17.38 <SkyCoord (ICRS): (ra, dec) in deg
    (313.523705, 37.021495)>
10:12:45.3 -45:17:50 <SkyCoord (ICRS): (ra, dec) in deg
    (153.1889, -45.2973)>
15h17m -11d10m <SkyCoord (ICRS): (ra, dec) in deg
    (229.25, -11.16666667)>
15h17 +89d15 <SkyCoord (ICRS): (ra, dec) in deg
    (229.25, 89.25)>
275d11m15.6954s +17d59m59.876s <SkyCoord (ICRS): (ra, dec) in deg
    (275.18769317, 17.99996556)>
12.34567h -17.87654d <SkyCoord (ICRS): (ra, dec) in deg
    (185.18505, -17.87654)>
350.123456d -17.33333d <SkyCoord (ICRS): (ra, dec) in deg
    (350.123456, -17.33333)>
350.123456 -17.33333 <SkyCoord (ICRS): (ra, dec) in deg
    (350.123456, -17.333333)>
```

IvanStrakhov committed:
get_coord_from_string added, try catch revised

IvanStrakhov commented:
Да, действительно. Я не знал об этой функции: SkyCoord.from_name(s, parse=True).
Я заменил свою функцию на вашу, с небольшим дополнением. Оно обусловлено (по крайней мере у меня) неправильным переводом координат (или вообще ошибкой) для случая, например, get_coord_from_string("123.0 12.0") или get_coord_from_string("123 12").

Блоки try catch пересмотрел и сократил.

hombit commented on Strakhov/requirements.txt:
```diff
@@ -0,0 +1,8 @@
+numpy
+Shapely
+astropy
+skipy
```
- scipy?
- Не хватает astroquery

hombit reviewed (changes requested):
Все отлично! Поправьте зависимости и готово

IvanStrakhov committed:
Update requirements.txt

IvanStrakhov commented:
Исправил

IvanStrakhov committed:
Update README.md
